pdr_backend/lake/table_bronze_pdr_slots.py

```diff
@@ -0,0 +1,161 @@
+from typing import Dict
+
+import polars as pl
+from enforce_typing import enforce_types
+from polars import Boolean, Float64, Int64, Utf8
+
+from pdr_backend.lake.plutil import (
+    pick_df_and_ids_on_period,
+)
+from pdr_backend.ppss.ppss import PPSS
+
+
+bronze_pdr_slots_table_name = "bronze_pdr_slots"
+
+# CLEAN & ENRICHED PREDICTOOR SLOTS SCHEMA
+bronze_pdr_slots_schema = {
+    "ID": Utf8,  # f"{contract}-{slot}"
+    "contract": Utf8,  # f"{contract}"
+    "pair": Utf8,
+    "timeframe": Utf8,
+    "source": Utf8,
+    "slot": Int64,
+    "roundSumStakesUp": Float64,
+    "roundSumStakes": Float64,
+    "trueval": Boolean,
+    "timestamp": Int64,
+    "last_event_timestamp": Int64,
+}
+
+
+def _process_slots(
+    collision_ids: pl.Series, dfs: Dict[str, pl.DataFrame], ppss: PPSS
+) -> Dict[str, pl.DataFrame]:
+    """
+    @description
+        Perform post-fetch processing on the data.
+        1. Find slots within the update
+        2. Transform slots to bronze
+        3. Concat to existing table
+    """
+
+    # only add new slots
+    slots_df = dfs["pdr_slots"].filter(
+        (pl.col("timestamp") >= ppss.lake_ss.st_timestamp)
+        & (pl.col("timestamp") <= ppss.lake_ss.fin_timestamp)
+        & (pl.col("ID").is_in(collision_ids).not_())
+    )
+
+    # select only the columns that we need
+    columns_to_select = [
+        "ID",
+        "slot",
+        "timestamp",
+        "roundSumStakesUp",
+        "roundSumStakes",
+    ]
+    slots_df = slots_df.select(columns_to_select)
+
+    # create contract column from the ID column
+    slots_df = slots_df.with_columns(
+        contract=pl.col("ID").map_elements(lambda s: s.split("-")[0])
+    )
+
+    if len(slots_df) == 0:
+        return dfs
+
+    if dfs[bronze_pdr_slots_table_name].schema == bronze_pdr_slots_schema:
```

Comment:
I don't know what this conditional statement is doing... as far as I can tell, it's a bug/extra...